### DISCUSSION TITLE
SELinux: Curl can't resolv host

### DISCUSSION THREAD

**Discussion by Marcs1987** [FreshRSS/FreshRSS#5875] [Q&A] [answered]
So I've installed FreshRSS on a AlmaLinux9 server. All the checks are green and I've imported a set of feeds that I follow. But the thing is that they don't update. And when I go the logs and look there a whole bunch of error messages like this one

cURL error 6: Could not resolve host: github.com [https://github.com/FreshRSS/FreshRSS/releases.atom] 

I tried 'sudo -u apache curl' from the command line and that works fine and I haven't found any other part that had problems resolving hostnames. 

Any ideas what's wrong?

**Comment by Alkarex** [comment]
What Web server?
Check maybe https://github.com/FreshRSS/FreshRSS/issues/2174

**Reply by Marcs1987** [reply]
I'm running Apache. Not really sure how to apply the information in #2174 to my case

**Comment by Alkarex** [comment]
Try from your command line:

```sh
php -r "readfile('http://freshrss.org/');"
```

And try also to make a PHP file with the following content, place it somewhere in your Web root, and call it from your browser:

```php
<?php
$ch = curl_init('http://freshrss.org/');
curl_setopt_array($ch, [
		CURLOPT_TIMEOUT => 15,
		CURLOPT_ENCODING => '',	//Enable all encodings
	]);
curl_exec($ch);
echo curl_errno($ch), ' (', curl_strerror(curl_errno($ch)), '). ', curl_error($ch);
curl_close($ch);
```

**Reply by Marcs1987** [reply]
Thanks for the suggestions

`php -r "readfile('http://freshrss.org/');"`

That worked fine.

Putting the suggested php file in the web root caused an error. Got the following in my browser
6 (Couldn't resolve host name). Could not resolve host: freshrss.org

**Reply by Alkarex** [reply]
Check maybe:
 
* https://superuser.com/questions/650000/cli-curl-works-flawlessly-apache-with-mod-php-cannot-resolve-host-how-to-fix
* https://superuser.com/questions/163256/php-curl-fedora-issues-resolving-hostname-possibly-a-dns-problem

**Comment by Marcs1987** [comment]
Thanks for the help.

It turned out to be things that Selinux blocked, after a bit of troubleshooting I managed to get it fixed.

**Reply by Alkarex** [reply]
Ah yes, if you use SELinux, that is the main reason for things not working out of the box, and this is by design of SELinux.
Could you consider improving our FAQ about SELinux, in English and/or French?

English:
* https://freshrss.github.io/FreshRSS/en/users/07_Frequently_Asked_Questions.html#permissions-under-selinux
* https://github.com/FreshRSS/FreshRSS/blob/edge/docs/en/users/07_Frequently_Asked_Questions.md

French:
* https://freshrss.github.io/FreshRSS/fr/users/07_Frequently_Asked_Questions.html#g%C3%A9rer-les-permissions-sous-selinux
* https://github.com/FreshRSS/FreshRSS/blob/edge/docs/fr/users/07_Frequently_Asked_Questions.md

**Comment by Marcs1987** [comment, marked as answer]
I'm afraid I didn't keep exact record of the changes I did. But basically I did the following

- Installed the packages selinux-policy, selinux-policy-devel, setroubleshoot-server and setroubleshoot (not installed by default)
- After that Selinux would log whenever it prevented some action together with instruction what commands to give if I wanted it to be allowed. So I would use the FreshRSS web interface and follow the instructions from the Selinux log to allow the prevented actions.
